code/modules/clothing/glasses/engine_goggles.dm

```diff
@@ -0,0 +1,76 @@
+//Engineering Mesons
+
+/obj/item/clothing/glasses/meson/engine
+	name = "Emgineering Meson Scanner"
+	desc = "Goggles used by engineers. The Meson Scanner mode lets you see basic structural and terrain layouts through walls, regardless of lighting condition. The T-ray Scanner mode lets you see underfloor objects such as cables and pipes."
+	icon_state = "meson"
+	item_state = "glasses"
+	origin_tech = "magnets=2;engineering=2"
+	darkness_view = 1
+	vision_flags = SEE_TURFS
+	invis_view = SEE_INVISIBLE_MINIMUM
+	var/state = 0	//0 - regular mesons mode	1 - t-ray mode
+	var/invis_objects = list()
+	action_button_name = "Change scanning mode"
+
+/obj/item/clothing/glasses/meson/engine/attack_self(mob/user)
+	ui_action_click()
+
+/obj/item/clothing/glasses/meson/engine/ui_action_click(mob/user)
+	state = !state
+	if(state)
+		SSobj.processing.Add(src)
```

Comment:
Can you use `SSobj.processing |= src` instead? If I spamclick the glasses I can probably add it more than once to the obj processing list which is bad.